http_backend.go

```diff
@@ -185,6 +187,12 @@ func (h *httpBackend) Do(request *http.Request, bodySize int) (*Response, error)
 	if bodySize > 0 {
 		bodyReader = io.LimitReader(bodyReader, int64(bodySize))
 	}
+	if res.Header.Get("Content-Encoding") == "gzip" {
```

Comment:
As I know golang's http lib handles compression by default. If it successfully decompresses the content, then `res.Uncompressed` [1] is `true`, so we could check here if decompression happened to avoid double decompression:

```go
if !res.Uncompressed && res.Header.Get("Content-Encoding") == "gzip" {
```

[1] https://golang.org/pkg/net/http/#Response